File: guides/building-a-rest-api-spring-boot-vs-micronaut-data/micronautframework/java/src/main/java/example/micronaut/SaasSubscription.java
package example.micronaut;

import io.micronaut.serde.annotation.Serdeable;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@Serdeable // <1>
@MappedEntity // <2>
record SaasSubscription(@Id Long id, // <3>
                        String name,
                        Integer cents) {
}

File: guides/building-a-rest-api-spring-boot-vs-micronaut-data/micronautframework/kotlin/src/main/kotlin/example/micronaut/SaasSubscription.kt
package example.micronaut

import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.serde.annotation.Serdeable

@Serdeable // <1>
@MappedEntity // <2>
data class SaasSubscription(
    @field:Id val id: Long, // <3>
    val name: String,
    val cents: Int
)
